src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -492,6 +500,82 @@ internal static string FullyQualifiedName(IEdmVocabularyAnnotatable element)
             return null;
         }
 
+        internal static string FullyQualifiedName(IEdmAnnotationsTarget annotationsTarget)
+        {
+            string qualifiedName = string.Empty;

```

Comment:
Is there a reason you can't initialize this variable at the time of assignment?

Reply:
Yes, If the input 'targetPath' is valid (not null), so as you did, it's a valid target path, the first segment is verified at its constructor